File: M1/C/buggy-source-fun.c
#include <stdio.h>

int inc(int i);
void pretty_print(int i);
int add(int i, int j);

int inc(int i) {
    return i + 1;
}

void pretty_print(int i) {
    printf("Here is %d\n", i);
}

int add(int i, int j) {
    return i + j;
}

int main(void) {
    int    i = 1;
    double j = 3.5;
    int    z = 0;

    pretty_print(i);

    printf("inc(%d) = %d\n", i, inc(i));

    printf("Now, an addition!\n");

    z = add(i, j);
    pretty_print(z);

    return 0;
}

File: M1/Rust/src/bin/buggy-source-fun.rs
fn incr(i: i32) -> i32{
    return i + 1;
}

fn pretty_print(i: i32) {
    println!("Here is {}", i);
}

fn add(i: f64, j: f64) {
    println!("{}", i + j);
}

fn main() {
    let i: i32 = 1;
    let j: f64 = 3.5;

    pretty_print(i);

    println!("inc({}) = {}", i, incr(i));

    println!("Now, an addition!");

    add(i as f64, j);
}
